core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -67,27 +67,231 @@ export class ReportGenerationService {
   }
 
   /**
-   * Generates an HTML file containing the workflow snapshot and triggers a download of the file.
-   * @param {string} workflowSnapshotURL - The base64-encoded PNG image URL of the workflow snapshot.
-   * @param {string} workflowName - The name of the workflow.
+   * Collects all operator results from the workflow and generates a comprehensive HTML report.
+   * @param {Object} payload - The payload containing operator IDs, operator details, workflow snapshot, and workflow name.
+   * @param {string[]} payload.operatorIds - Array of operator IDs present in the workflow.
+   * @param {any[]} payload.operators - Array of operator details, including configurations and metadata.
+   * @param {string} payload.workflowSnapshot - Base64-encoded PNG image URL of the workflow snapshot.
+   * @param {string} payload.workflowName - The name of the workflow, used for naming the final report.
+   * @returns {void}
+   */
+
+  public getAllOperatorResults(payload: {
+    operatorIds: string[];
+    operators: any[];
```

Comment:
avoid using `any`. Define a class containing the field you want to show in the report.

the `operatorIds` here is redundant. You can put the id in `operators` parameter. 

rename `workflowSnapshot` to `workflowSnapshotURL`